### PR TITLE
LLM Validator

### PR DESCRIPTION
**Pull request by ethanleifer**
Using Zod's [`superRefine`](https://zod.dev/?id=superrefine) schema validator: 

Created a function `LLMValidator` that returns an asynchronous function that uses an instructor instance to check the schema output against a statement

How do we want to this into the docs @jxnl @roodboi?

Few notes:
- Switched to asyncParse where needed
- Used `superRefine` instead of `transform` as in #25 since we only want to return true/false and let the base instructor handle editing

### CODE REVIEW

**Comment by roodboi** [comment]
cookbook + example maybe?

**Comment by ethanleifer** [comment]
> cookbook + example maybe?

Added example
Added Cookbook
Added Self Correction Test

File: docs/examples/self-correction.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
+# Self-Correction with `LLMValidator`
```
Comment (jxnl) [inline docs/examples/self-correction.md:1]: make sure to add it to the mkdocs.md if it does not exist

**Review by jxnl — APPROVED** [review]
going to accept but please add the mkdoc ref if it does not exist

**Pull request merged**